quixstreams/dataframe/registry.py

```diff
@@ -89,3 +96,39 @@ def compose_all(
             root_executors = root_stream.compose(sink=sink)
             executors[topic] = root_executors[root_stream]
         return executors
+
+    def register_stream_id(self, stream_id: str, topic_names: list[str]):
+        """
+        Register a mapping between the stream_id and topic names.
+        This mapping is later used to match topics to state stores
+        during assignment and commits.
+
+        The same stream id can be registered multiple times.
+         # TODO: Test
+        :param stream_id: stream id of StreamingDataFrame
+        :param topic_names: list of topics to map the stream id with
+        """
+        for topic_name in topic_names:
+            self._topics_to_stream_ids.setdefault(topic_name, set()).add(stream_id)
+            self._stream_ids_to_topics.setdefault(stream_id, set()).add(topic_name)
+
+    def get_stream_ids(self, topic_name: str) -> list[str]:
+        """
+        Get a list of stream ids for the given topic name
+        # TODO: Test
+
+        :param topic_name: a name of the topic
+        :return: a list of stream ids
+        """
+        return list(self._topics_to_stream_ids[topic_name])
+
+    def get_topics_for_stream_id(self, stream_id: str) -> list[str]:
```

Comment:
Having `stream_id` as an argument is self-explanatory.
```suggestion
    def get_topics(self, stream_id: str) -> list[str]:
```

Reply:
I'd keep the more specific `get_topics` because the public property `consumer_topics` is in the same class.